simulator.py

```diff
@@ -0,0 +1,123 @@
+# Class - Simulator
+# Runs multiple simulations of the game and saves the results
+
+import game
+import concurrent.futures
+import csv
+
+class Simulator:
+    """Class that manages simulations of a BlackJack game
+    
+    Parameters:
+
+    sim_count (int): Indicates the number of simulations to compute - will be rounded up to the
+    
+    nearest multiple of 100 
+    
+    bet (int): Represents the starting bet for the Black Jack game
+    """
+
+    def __init__(self, sim_count=1000000, bet_amount=1):
+        if sim_count % 10000 != 0: 
+            self.sim_count = sim_count + 10000 - (sim_count % 10000)
+        else:
+            self.sim_count = sim_count
+
+        self.bet_amount = bet_amount
+        self.completed_sims = 0
+
+    def start(self):
+        """ Runs the Simulation with the initialized attributes.
+
+        Outputs:
+        
+        The result of the current simulation as well as the overall result of all simulations run previously.
+
+        Appends the most recent result to results.csv
+        
+        Updates overall_avg.csv with the updated totals for total_sims and result_sum
+        """
+        print(f'Starting Blackjack Simulator with {self.sim_count} simulations')
+        results = []
+
+        with concurrent.futures.ProcessPoolExecutor() as executor:
+            p_list = []
+            for i in range(int(self.sim_count/10000)):
+                p_list.append(executor.submit(self.run_multiple_games, 10000))
+                
+                for res in concurrent.futures.as_completed(p_list):
+                    results.append(res.result())
```

Comment:
It doesn't seem any faster.. maybe I'm missing something?

Reply:
I think the second for-loop at line 48 shouldn't be nested since it's looking at the list of processes after they have spawned/started.

Reply:
Tried doing some testing of my own with the Python concurrent stuff.
Like you said it doesn't seem to actually improve our times.